src/utils/actions.js

```diff
@@ -0,0 +1,254 @@
+// in turn.  I'm also assuming the context is unique
+// to the current action, since we'd be cycling through actions
+// anyway.
+const Fhir = require('fhir').Fhir;
+const FhirPath = require('fhir').FhirPath;
+const fhir = new Fhir();
+const baseIgActions = {
+  'check-trigger-codes': context => {
+    const action = context.action;
+    const resources = context.records;
+
+    const boolMap = action.condition.map(condition => {
+      if (condition.kind === 'applicability' && condition.expression) {
+        const expression = condition.expression;
+        // check the records for a trigger code
+        return resources.some(resource => {
+          //might have to check that this is non-empty here
+          return evaluateExpression(expression, resource);
+        });
+      } else {
+        // default true for non-applicability conditions
+        return true;
+      }
+    });
+    // boolMap should be all-true (all conditions met)
+    const triggered = boolMap.every(entry => {
+      return entry === true;
+    });
+    context.flags['triggered'] = triggered;
+  },
+  'check-participant-registration': context => {
+    return null;
+  },
+  'create-report': context => {
+    // the output might define a profile
+    let profile = null;
+    if (context.action.output) {
+      const output = context.action.output[0];
+      if (output.profile) {
+        profile = output.profile[0];
+      }
+    }
+    let contentBundle = createBundle(context.records);
+    if (profile) {
+      contentBundle = validateProfile(contentBundle, profile);
+    }
+    const header = makeHeader();
+    context.contentBundle = contentBundle;
+    context.reportingBundle = createBundle([header, ...contentBundle.entry]);
+  },
+  'validate-report': context => {
+    // check if fhir is well-formed
+    const validation = fhir.validate(context.reportingBundle);
+    context.flags['valid'] = validation.valid;
+  },
+  'submit-report': context => {
+    context.client.dest.submit(context.reportingBundle);
+  },
+  'deidentify-report': context => {
+    const client = context.client.trustServices;
+    client.deidentify(context.reportingBundle).then(result => {
+      context.reportingBundle = result;
+    });
+    context.flags['deidentified'] = true;
+  },
+  'anonymize-report': context => {
+    const client = context.client.trustServices;
+    client.anonymize(context.reportingBundle).then(result => {
+      context.reportingBundle = result;
+    });
+    context.flags['anonymized'] = true;
+  },
+  'pseudonymize-report': context => {
+    const client = context.client.trustServices;
+    client.pseudonymize(context.reportingBundle).then(result => {
+      context.reportingBundle = result;
+    });
+    context.flags['pseudonymized'] = true;
+  },
+  'encrypt-report': context => {
+    // noop
+    context.flags['encrypted'] = true;
+  },
+  'complete-reporting': context => {
+    context.client.db.insert('completed reports', context.reportingBundle);
```

Comment:
Since this is a new collection which isn't a FHIR type could you please add it to the [Collections wiki](https://github.com/mcode/medmorph-backend/wiki/Collections)